Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/MousePointer.cs

```diff
@@ -207,67 +72,44 @@ public override void OnInputChanged(InputEventData<MixedRealityPose> eventData)
             }
         }
 
-        #endregion IMixedRealityInputHandler Implementation
-
-        #region MonoBehaviour Implementation
-
-        protected override void Start()
+        private void UpdateMouseRotation(Vector3 mouseDeltaRotation)
         {
-            isDisabled = DisableCursorOnStart;
-
-            base.Start();
-
-            if (RayStabilizer != null)
+            if (isDisabled)
             {
-                RayStabilizer = null;
-            }
-
-            foreach (var inputSource in InputSystem.DetectedInputSources)
-            {
-                if (inputSource.SourceId == Controller.InputSource.SourceId)
+                if (mouseDeltaRotation.magnitude >= MovementThresholdToUnHide)
                 {
-                    isInteractionEnabled = true;
-                    break;
+                    // if cursor was hidden reset to center
+                    SetVisibility(true);
+                    transform.rotation = CameraCache.Main.transform.rotation;
                 }
             }
-        }
-
-        private void Update()
-        {
-            if (!hideCursorWhenInactive || isDisabled) { return; }
-
-            timeoutTimer += Time.unscaledDeltaTime;
-
-            if (timeoutTimer >= hideTimeout)
+            else
             {
                 timeoutTimer = 0.0f;
-                BaseCursor?.SetVisibility(false);
-                isDisabled = true;
             }
-        }
 
-        #endregion MonoBehaviour Implementation
+            transform.Rotate(mouseDeltaRotation, Space.World);
+        }
 
-        private void UpdateMouseRotation(Vector3 mouseDeltaRotation)
+        protected override void Start()
         {
-            if (mouseDeltaRotation.magnitude >= movementThresholdToUnHide)
-            { 
-                if (isDisabled)
-                {
-                    // if cursor was hidden reset to center
-                    BaseCursor?.SetVisibility(true);
-                    transform.rotation = CameraCache.Main.transform.rotation;
-                }
-
-                isDisabled = false;
-            }
+            base.Start();
 
-            if (!isDisabled)
+#if UNITY_EDITOR
+            if (UnityEditor.EditorWindow.focusedWindow != null)
             {
-                timeoutTimer = 0.0f;
+                UnityEditor.EditorWindow.focusedWindow.ShowNotification(new GUIContent("Press \"ESC\" to regain mouse control"));
             }
+#endif
 
-            transform.Rotate(mouseDeltaRotation, Space.World);
+            Cursor.visible = false;
+            Cursor.lockState = CursorLockMode.Locked;
+        }
+
+        protected override void SetVisibility(bool visible)
```

Comment:
do you need this override? isn't the behavior the same as in the base class?

Reply:
was partially different, now it's more clear